src/librc/librc-depend.c

```diff
@@ -22,7 +22,17 @@
 
 #define GENDEP          RC_LIBEXECDIR "/sh/gendepends.sh"
 
-#define RC_DEPCONFIG    RC_SVCDIR "/depconfig"
+#define RC_DEPCONFIG    "/depconfig"
+
+static char *configpath = NULL;
+static char*
+depconfigpath_get() {
+	if (!configpath) {
+		configpath = xmalloc(sizeof(char) * PATH_MAX);
+		snprintf(configpath, PATH_MAX, "%s/%s", rc_svcdir_get(), RC_DEPCONFIG);
```

Comment:
`PATH_MAX` is bit of a [nonsensical constant](https://eklitzke.org/path-max-is-tricky), openrc uses it already is a decent amount of places, but I'd prefer not to introduce it to new code.

Reply:
What would be better than using PATH_MAX? I knew about it's issues, but didn't about a better solution...

Reply:
Look into some of the `xasprintf` usage. It will allocate as much as needed.